sway-lib-std/src/math.sw

```diff
@@ -193,15 +211,65 @@ impl BinaryLogarithm for u8 {
     }
 }
 
+impl BinaryLogarithm for u256 {
+    /// Calculates the binary log.
+    /// 
+    /// Decompose to u64 and choose the most significant bits
+    /// among the four u64s.
+    /// 
+    /// # Panics
+    ///
+    /// Panics if the operand is 0.
+    fn log2(self) -> Self {
+        use ::assert::*;
+        assert(self != 0);
+        let (a,b,c,d) = asm(r1: self) {
+            r1: (u64, u64, u64, u64)
+        };
+        if a != 0 {
+            return a.log2().as_u256() + 0xc0u256;
+        } else if b != 0 {
+            return b.log2().as_u256() + 0x80u256;
+        } else if c != 0 {
+            return c.log2().as_u256() + 0x40u256;
+        } else if d != 0 {
+            return d.log2().as_u256();
+        }
+        self
+    }
+}
+
+impl Logarithm for u256 {
```

Comment:
Please move this implementation above the implementation for u64

Reply:
Same as the other one - `log` depends on `log2` for `u256` and hence should be after `log2`
```
error
   --> /home/sudhackar/personal/sway/sway-lib-std/src/math.sw:152:30
    |
150 |
151 |     fn log(self, base: Self) -> Self {
152 |         let self_log2 = self.log2();
    |                              ^^^^ No method named "log2" found for type "u256".
153 |         let base_log2 = base.log2();
154 |         self_log2 / base_log2
    |
____

```